core/src/test/java/org/bitcoinj/crypto/HDPathTest.java

```diff
@@ -58,6 +58,22 @@ public void testExtendVarargs() {
         assertEquals("m/0H/1H/0H/1/0",  path5.toString());
     }
 
+    @Test
+    public void testParent() {
+        HDPath path1 = HDPath.parsePath("m/0H/1H");
+
+        assertEquals(HDPath.parsePath("m/0H"), path1.parent());
+
+        HDPath path2 = HDPath.parsePath("m/0H");
+
+        assertEquals(HDPath.parsePath(""), path2.parent());
```

Comment:
I'm not too familiar with these paths, but should this parent not be 'm'?

Reply:
It's actually an empty path at that point. The string representation is a little confusing. But thing of the underlying `List<ChildNumber>`. That is an empty list.

Reply:
Ok, agreed.